### PR TITLE
fix(ivy): update content query compilation to latest runtime

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
This PR updates compiler to generate content queries code expected by the runtime.

### CODE REVIEW

File: packages/compiler/src/render3/r3_factory.ts [hunk 1]
```diff
@@ -155,12 +155,8 @@ export function compileFactoryFunction(meta: R3FactoryMetadata): o.Expression {
   const expr = meta.useNew ? new o.InstantiateExpr(meta.fnOrClass, args) :
                              new o.InvokeFunctionExpr(meta.fnOrClass, args);
 
-  // If `extraResults` is specified, then the result is an array consisting of the instantiated
-  // value plus any extra results.
-  const retExpr =
-      meta.extraResults === undefined ? expr : o.literalArr([expr, ...meta.extraResults]);
```
Comment (pkozlowski-opensource) [inline packages/compiler/src/render3/r3_factory.ts]: @alxhub I wonder if we can get rid of `extraResults` now or do you want to keep it around for other purposes?

**Review by mhevery — APPROVED** [review]
LGTM, @alxhub can you take a look as well?

File: packages/compiler/src/render3/r3_factory.ts (hunk 1 above)
Comment (alxhub) [reply packages/compiler/src/render3/r3_factory.ts]: We can get rid of it for sure.

**Review by alxhub — APPROVED** [review]
This will definitely conflict with #24862, which will merge soon.

**Comment by pkozlowski-opensource** [comment]
Marking as `merge-assistance`, since:
* re-approval is needed after pushing a fixup commit
* CircleCI fails for un-related reasons
* g3sync presubmit needs to be run

**Pull request closed without merging**